### PR TITLE
Raspberry Pi3 (ARM64) support

### PR DESCRIPTION
**Pull request by brainstorm**
As discussed on the Discord server, here you have some ARM64 action without docker nor github actions (reusing the existing TravisCI recipe and deploy keys), hope the build doesn't fail too much ;)

It works fine locally using docker on OSX:

```
$ docker build --tag openpnp-capture-rpi3-arm64:latest . -f docker/Dockerfile.rpi3_arm64
(...)
$ docker run -v $PWD:/openpnp-capture openpnp-capture-rpi3-arm64 /bin/bash -c "./bootstrap_linux.sh && cd build && make"
$ file build/libopenpnp-capture.so
build/libopenpnp-capture.so: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, BuildID[sha1]=23504b1c2819e17d67904b63f44d8abdc4d0d685, not stripped
```

### CODE REVIEW

**Comment by brainstorm** [comment]
Fixes https://github.com/openpnp/openpnp-capture/issues/27, btw :)

File: .travis.yml [hunk 1]
```diff
@@ -1,37 +1,60 @@
...
 
 script: make
+# docker run -v $PWD:/openpnp-capture openpnp-capture-rpi3-arm64 make
 
 before_deploy:
   - mkdir deploy
+  - if [ $TRAVIS_OS_NAME = windows ]; then cp libopenpnp-capture.dll deploy; cp linux/tests/openpnp-capture-test deploy/openpnp-capture-test-windows; fi
   - if [ $TRAVIS_OS_NAME = osx ]; then cp libopenpnp-capture.dylib deploy; cp mac/tests/openpnp-capture-test deploy/openpnp-capture-test-mac; fi
   - if [ $TRAVIS_OS_NAME = linux ]; then cp libopenpnp-capture.so deploy; cp linux/tests/openpnp-capture-test deploy/openpnp-capture-test-linux; fi
+  - if [ $TRAVIS_CPU_ARCH = arm64 ]; then cp libopenpnp-capture.so deploy; cp linux/tests/openpnp-capture-test deploy/openpnp-capture-test-linux-arm64; fi
```
Comment (vonnieda) [inline .travis.yml:50]: I think this is going to overwrite the Linux x64 build, since it has the same name. I'm going to test some changes to append the platform and arch to each binary to make everything clearer.

File: .travis.yml [hunk 2]
```diff
@@ -1,37 +1,60 @@
...
+    - libjpeg-dev
 
 before_script:
+#  - docker build --tag openpnp-capture-rpi3-arm64:latest . -f docker/Dockerfile.rpi3_arm64 
   - mkdir build
   - cd build
   - cmake ..
 
 script: make
+# docker run -v $PWD:/openpnp-capture openpnp-capture-rpi3-arm64 make
```
Comment (vonnieda) [inline .travis.yml:43]: Note that make won't work for Windows. See the appveyor.xml for info on how we build there. We may need to make changes for Travis.

File: .travis.yml [hunk 3]
```diff
@@ -1,37 +1,60 @@
...
 
 before_deploy:
   - mkdir deploy
+  - if [ $TRAVIS_OS_NAME = windows ]; then cp libopenpnp-capture.dll deploy; cp linux/tests/openpnp-capture-test deploy/openpnp-capture-test-windows; fi
```
Comment (vonnieda) [inline .travis.yml:47]: Depending on how we build (using Visual Studio, probably), I don't think this filename will be correct.

File: .travis.yml (hunk 2 above)
Comment (brainstorm) [reply .travis.yml:43]: Yeah, I know, my intention was to go for this: https://docs.travis-ci.com/user/reference/windows/#how-do-i-use-msys2 ... I've seen the appveyor.xml, first I wanted to get some red though, I'll trigger it in the next few minutes...

File: .travis.yml (hunk 3 above)
Comment (brainstorm) [reply .travis.yml:47]: Gotcha, we'll see what it generates after I put the windows bits in.

File: .travis.yml (hunk 1 above)
Comment (brainstorm) [reply .travis.yml:50]: IIRC concurrent job builds are executed in separate contexts/VMs/filesystems? The x86 build should not affect the arm64 one... I think, I might be wrong though :-S

Comment (brainstorm) [reply .travis.yml:51]: Ohh, I see what you mean now, overwritten on deploy time, of course! :-/

Comment (brainstorm) [reply .travis.yml:51]: Do you have the changes you refer to (append platform) in some other branch or staged locally? Let's ship this PR and refine the appveyor vs travis windows migration things on another PR? I want to get the RPi3 functionality shipped first ;)

Comment (vonnieda) [reply .travis.yml:51]: No - I started on it then got wrapped up in Windows stuff. If you just want to change it so that it won't overwrite on deploy we can go from there.

**Comment by vonnieda** [comment]
Thanks @brainstorm. If you have a moment, can you verify that the resulting binary does not have any external dependencies? It looks to me like it might end up requiring libjpeg installed by the user, which makes the binary likely to fail in confusing ways for users. You can check the external dependencies with `ldd library.so`. If it is using libjpeg (or anything else) as a shared dependency, check out the CMakeLists.txt to see how we fix this issue using turbojpeg.

**Comment by vonnieda** [comment]
There does seem to be some libjpeg issues here: https://travis-ci.org/github/openpnp/openpnp-capture/jobs/676039253

I'm suspicious about all three added dependencies:

```
    - build-essential
    - cmake
    - libjpeg-dev
```

Why were these needed?

**Comment by brainstorm** [comment]
**TL;DR: Sorry, yes, please, kill those 3 dependencies from `.travisci.yml`** ;)

Sure (OSX does not have `ldd`, so resorting to the docker container itself). There doesn't seem to be such an external dep:

```
$ docker run -it -v $PWD:/openpnp-capture openpnp-capture-rpi3-arm64 bash
root@b7d320bc76de:/openpnp-capture# ldd build/libopenpnp-capture.so
	libstdc++.so.6 => /usr/lib/aarch64-linux-gnu/libstdc++.so.6 (0x00000040008d8000)
	libm.so.6 => /lib/aarch64-linux-gnu/libm.so.6 (0x0000004000a65000)
	libgcc_s.so.1 => /lib/aarch64-linux-gnu/libgcc_s.so.1 (0x0000004000b22000)
	libpthread.so.0 => /lib/aarch64-linux-gnu/libpthread.so.0 (0x0000004000b46000)
	libc.so.6 => /lib/aarch64-linux-gnu/libc.so.6 (0x0000004000b75000)
	/lib/ld-linux-aarch64.so.1 (0x0000004000000000)
```

Plus they are explicitly disabled on CMakeLists, right?:

https://github.com/openpnp/openpnp-capture/blob/master/CMakeLists.txt#L121

You are totally right on the three deps. Those dependencies are not needed on `.travisci.yml` because the TravisCI VMs already ships with them. Otherwise you find things like this on bare docker containers or VMs (what the users will find):

```
$ docker run -it -v $PWD:/openpnp-capture openpnp-capture-rpi3-arm64 ./bootstrap_linux.sh
/bin/mkdir: cannot create directory ‘build’: File exists
/openpnp-capture/bootstrap_linux.sh: 4: /openpnp-capture/bootstrap_linux.sh: cmake: not found


$ docker run -it -v $PWD:/openpnp-capture openpnp-capture-rpi3-arm64 ./bootstrap_linux.sh
/bin/mkdir: cannot create directory ‘build’: File exists
CMake Error at CMakeLists.txt:14 (project):
  The CMAKE_C_COMPILER:

    /usr/bin/cc

  is not a full path to an existing compiler tool.

  Tell CMake where to find the compiler by setting either the environment
  variable "CC" or the CMake cache entry CMAKE_C_COMPILER to the full path to
  the compiler, or to the compiler name if it is in the PATH.


CMake Error at CMakeLists.txt:14 (project):
  The CMAKE_CXX_COMPILER:

    /usr/bin/c++

  is not a full path to an existing compiler tool.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/openpnp-capture/build/CMakeFiles/CMakeOutput.log".
See also "/openpnp-capture/build/CMakeFiles/CMakeError.log".
```

But you are spot on, they are not needed on this particular CI context and the error messages from TravisCI suggest that `libjpeg-dev` is interfering with newer `libjpeg-dev` headers and CMake/gcc.

**Comment by brainstorm** [comment]
This is a bit weird since my builds were pristine:

https://travis-ci.org/github/openpnp/openpnp-capture/jobs/676036339
https://travis-ci.org/github/openpnp/openpnp-capture/builds/676036337
